### PR TITLE
Display linear/multiplicity-polymorphic arrows correctly

### PR DESCRIPTION
**Pull request by monoidal**
Haddock was not showing linear and multiplicity-polymorphic arrows. I think this was a shortcut taken during implementation of linear types.

### CODE REVIEW

File: haddock-api/src/Haddock/Backends/LaTeX.hs [hunk 1]
```diff
@@ -1367,14 +1371,18 @@ quote :: LaTeX -> LaTeX
...
 dcolon unicode = text (if unicode then "∷" else "::")
 arrow  unicode = text (if unicode then "→" else "->")
+lollipop unicode = text (if unicode then "⊸" else "%1 ->")
 darrow unicode = text (if unicode then "⇒" else "=>")
 forallSymbol unicode = text (if unicode then "∀" else "forall")
 starSymbol unicode = text (if unicode then "★" else "*")
 atSign unicode = text (if unicode then "@" else "@")
 
+multAnnotation :: LaTeX
+multAnnotation = text "%"
```
Comment (goldfirere) [inline haddock-api/src/Haddock/Backends/LaTeX.hs:1384]: This looks suspect. Maybe `text "\\%"`?

Comment (alexbiehl) [reply haddock-api/src/Haddock/Backends/LaTeX.hs:1384]:
```suggestion
multAnnotation = text "\\%"
```

**Comment by alexbiehl** [comment]
This looks reasonable so far but could use a test.

File: haddock-api/src/Haddock/Backends/LaTeX.hs (hunk 1 above)
Comment (monoidal) [reply haddock-api/src/Haddock/Backends/LaTeX.hs:1384]: No, escaping already happens here:
```
latexMunge '%'  s = "{\\char '45}" ++ s
```

**Comment by monoidal** [comment]
I added a test - this is ready for another review.

**Comment by alexbiehl-gc** [comment]
Cool! Looking good!

**Review by alexbiehl — APPROVED** [review]

**Pull request merged**